File: frontend/src/modules/cruise-applications/models/CruiseDayDetailsDto.ts
import { z } from 'zod';

export type CruiseDayDetailsDto = {
  number: string;
  hours: string;
  taskName: string;
  region: string;
  position: string;
  comment: string;
};

export const CruiseDayDetailsDtoValidationSchema = z.object({
  number: z.string().nonempty('Numer dnia rejsu jest wymagany'),
  hours: z.string().nonempty('Godziny są wymagane'),
  taskName: z.string().nonempty('Nazwa zadania jest wymagana'),
  region: z.string().nonempty('Region jest wymagany'),
  position: z.string().nonempty('Pozycja jest wymagana'),
  comment: z.string().nonempty('Komentarz jest wymagany').max(10240, 'Komentarz nie może być dłuższy niż 10240 znaków'),
});

File: backend/ResearchCruiseApp/Api/Applications/Contracts/CruiseDayDetailsDto.cs
using System.ComponentModel.DataAnnotations;

namespace ResearchCruiseApp.Api.Applications.Contracts;

public class CruiseDayDetailsDto
{
    [StringLength(1024)]
    public string Number { get; init; } = null!;

    [StringLength(1024)]
    public string Hours { get; init; } = null!;

    [StringLength(1024)]
    public string TaskName { get; init; } = null!;

    [StringLength(1024)]
    public string Region { get; init; } = null!;

    [StringLength(1024)]
    public string Position { get; init; } = null!;

    [StringLength(1024)]
    public string Comment { get; init; } = null!;
}
